diff-java-tool/src/main/java/com/github/checkstyle/difftool/DiffTool.java

```diff
@@ -0,0 +1,1115 @@
+            Files.walkFileTree(directory, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                    Files.delete(file);
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                    Files.delete(dir);
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        }
+    }
+
+    private static void executeCmd(String cmd, File dir) throws IOException, InterruptedException {
+        System.out.println("Running command: " + cmd);
+        ProcessBuilder pb = new ProcessBuilder(getOsSpecificCmd(cmd).split("\\s+"));
+        pb.directory(dir);
+        pb.inheritIO();
+        Process process = pb.start();
+        int exitCode = process.waitFor();
+        if (exitCode != 0) {
+            throw new RuntimeException("Error: Command execution failed with exit code " + exitCode);
+        }
+    }
+
+    private static void executeCmd(String cmd) throws IOException, InterruptedException {
+        executeCmd(cmd, new File("").getAbsoluteFile());
+    }
+
+    private static String getOsSpecificCmd(String cmd) {
+        if (System.getProperty("os.name").toLowerCase().contains("windows")) {
+            return "cmd /c " + cmd;
+        }
+        return cmd;
+    }
+
+    private static String getOsSpecificPath(String... name) {
+        return String.join(File.separator, name);
+    }
+
+    private static boolean isWindows() {
+        return System.getProperty("os.name").toLowerCase().contains("windows");
+    }
+
+    private static String getResetCmd(String repoType, String commitId) {
+        if ("git".equals(repoType)) {
+            if (isGitSha(commitId)) {
+                return "git reset --hard " + commitId;
+            } else {
+                return "git reset --hard refs/tags/" + commitId;
+            }
+        }
+        throw new IllegalArgumentException("Error! Unknown " + repoType + " repository.");
+    }
+
+    private static String getLastProjectCommitSha(String repoType, String srcDestinationDir) throws IOException, InterruptedException {
+        if ("git".equals(repoType)) {
+            ProcessBuilder pb = new ProcessBuilder("git", "rev-parse", "HEAD");
+            pb.directory(new File(srcDestinationDir));
+            Process process = pb.start();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String sha = reader.readLine();
+                process.waitFor();
+                return sha != null ? sha.trim() : "";
+            }
+        }
+        throw new IllegalArgumentException("Error! Unknown " + repoType + " repository.");
+    }
+
+
+    // ... (other methods will follow)
```

Comment:
What is left?

Reply:
As soon as we convert all, we can build jar, put it to releases and let checkstyle workflow simply download jar without any mess with cloning repo and building from scratch.

Reply:
Done. Nothing, as you know i had to take help of gpt for conversion. So, it remained as a comment which i forgot to edit.